EasyFinance.Domain/AccessControl/UserProject.cs

```diff
@@ -9,17 +9,29 @@ public class UserProject : BaseEntity
     {
         private UserProject() { }
 
-        public UserProject(User user = default, Project project = default, Role role = default) 
+        public UserProject(User? user = default,  Project project = default, Role role = default, string? email = default)
         {
-            this.SetUser(user ?? new User());
+            if (user == null && string.IsNullOrEmpty(email))
+                throw new ValidationException("Either a User or an Email must be provided for the UserProject.");
+
+            this.User = user;
+            this.Email = email;
             this.SetProject(project ?? new Project());
             this.SetRole(role);
         }
 
-        public User User { get; private set; } = new User();
+        public User? User { get; private set; } 
+        public string? Email { get; set; }      
         public Project Project { get; private set; } = new Project();
         public Role Role { get; private set; }
 
+        public Guid Token { get; set; }     
+        public bool Accepted { get; set; }        
+        public DateTime SentAt { get; set; }       
+        public DateTime? AcceptedAt { get; set; }
+        public bool Expired { get; set; } 
+        public DateTime ExpiryDate { get; set; } 
```

Comment:
```suggestion
        public Guid Token { get; private set; }     
        public bool Accepted { get; private set; }        
        public DateTime SentAt { get; private set; }       
        public DateTime? AcceptedAt { get; private set; }
        public bool Expired { get; private set; } 
        public DateTime ExpiryDate { get; private set; } 
```

All the models in this project has private set and you need to create a method to add the value and in this method before add the new value you will validate if the value is valid. The objetive here is use strong domains that keep valid eachself. 

You can also add in contructor each value to be added in the creation of the model calling this internal method to validate the values.

If you need some example you can look any model in this project.